clang/labs/lab1/travin_maksim/TravinAlwaysInline.cpp

```diff
@@ -0,0 +1,72 @@
+#include "clang/AST/ASTContext.h"
+#include "clang/AST/Attr.h"
+#include "clang/AST/RecursiveASTVisitor.h"
+#include "clang/AST/Stmt.h"
+#include "clang/Frontend/CompilerInstance.h"
+#include "clang/Frontend/FrontendPluginRegistry.h"
+
+class InlineVisitor : public clang::RecursiveASTVisitor<InlineVisitor> {
+private:
+  clang::ASTContext *Context;
+  bool HasStatement;
+
+public:
+  explicit InlineVisitor(clang::ASTContext *Context)
+      : Context(Context), HasStatement(false) {}
+
+  bool VisitFunctionDecl(clang::FunctionDecl *Func) {
+    HasStatement = false;
+    TraverseStmt(Func->getBody());
+    if (Func->hasAttr<clang::AlwaysInlineAttr>()) {
+      Func->dump();
```

Comment:
Is it necessary? If it is already there then `-ast-dump` option should print it anyway.

Reply:
No, it's not necessary